warehouse/azure-synapse/azure-synapse.go

```diff
@@ -854,3 +854,7 @@ func (as *HandleT) LoadTestTable(location string, tableName string, payloadMap m
 func (as *HandleT) SetConnectionTimeout(timeout time.Duration) {
 	as.ConnectTimeout = timeout
 }
+
+func (*HandleT) RefreshPartitions(_ string, _ []warehouseutils.LoadFileT) (err error) {
```

Comment:
It's not related to the corresponding changes. It's something in General. 
I don't think we should expose this to the WareHouse Manager since on DataLake is only implementing this and others are not implementing this. 
I guess, we can opt for some other design pattern here for these kind of changes going forward.

Reply:
@achettyiitr That's a great comment. It makes sense for only datalakes to implement this. A separate interface for this would make more sense.